lib/schema.js

```diff
@@ -348,6 +348,7 @@ Schema.prototype.clone = function() {
   s.callQueue = this.callQueue.map(function(f) { return f; });
   s.methods = utils.clone(this.methods);
   s.statics = utils.clone(this.statics);
+  s.s.hooks = utils.clone(this.s.hooks);
```

Comment:
The more correct way of doing this is `this.s.hooks.clone()` but either way works. Thanks for finding this :+1: